### PR TITLE
add methods to R client

### PR DESCRIPTION
**Pull request by fostermh**
update httr to httr2 library. add  http post, put, and patch methods

### CODE REVIEW

File: hakaiApi/DESCRIPTION [hunk 1]
```diff
@@ -19,6 +19,7 @@ Depends:
 Imports:
     dplyr,
     httr,
+    httr2,
```
Comment (tayden) [inline hakaiApi/DESCRIPTION:22]: Is `httr` still required with the addition of `httr2`? Can we get ride of `httr`?

**Review by tayden — APPROVED** [review]
Looks good. Would be nice to drop httr if we can and use only httr2.

I'll have to remember how to get through R's obnoxious package approval process to push this to their package hub. For now, everyone can go back to installing via github.

File: hakaiApi/DESCRIPTION (hunk 1 above)
Comment (fostermh) [reply hakaiApi/DESCRIPTION:22]: httr is not required.

Comment (fostermh) [reply hakaiApi/DESCRIPTION]: and yes I was also looking at how to publish it. something about uploading a tar.gz  via a web form?

**Pull request merged**